File: OrangeJuice.Server.Core/Configuration/AzureOptionsFactory.cs
namespace OrangeJuice.Server.Configuration
{
	public class AzureOptionsFactory : Factory.IFactory<AzureOptions>
	{
		private readonly IConfigurationProvider _configurationProvider;

		public AzureOptionsFactory(IConfigurationProvider configurationProvider)
		{
			_configurationProvider = configurationProvider;
		}

		public AzureOptions Create()
		{
			return new AzureOptions
			{
				ConnectionString = _configurationProvider.GetValue("azure:Blob"),
				ProductsContainer = _configurationProvider.GetValue("azure:container:Products"),
				AwsOptionsTable = _configurationProvider.GetValue("azure:table:AwsOptions"),
			};
		}
	}
}

File: OrangeJuice.Server.Core.FSharp/Configuration/AzureOptionsFactory.fs
namespace OrangeJuice.Server.FSharp.Configuration

open OrangeJuice.Server.Configuration

type AzureOptionsFactory(configurationProvider : IConfigurationProvider) = 
    interface Factory.IFactory<AzureOptions> with
        member this.Create() : AzureOptions = 
            new AzureOptions(ConnectionString = configurationProvider.GetValue("azure:Blob"), 
                             ProductsContainer = configurationProvider.GetValue("azure:container:Products"), 
                             AwsOptionsContainer = configurationProvider.GetValue("azure:container:AwsOptions"))
